### PR TITLE
Intercom: user id bug

### PR DESCRIPTION
**Pull request by shrouti1507**
## Description of the change
Now the user has control, whether they want to send anonymousId as secondary userId, which we are planning to use as user_id in the output payload if userId and email are missing from input.

https://www.notion.so/rudderstacks/Intercom-UserId-bug-fix-0243d0386e5e4e8cb7111ce1353bf379
## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/intercom/transform.js [hunk 1]
```diff
@@ -143,6 +144,9 @@ function processSingleMessage(message, destination) {
 
   // build the response and return
   const payload = constructPayload(message, MappingConfig[category.name]);
+  if (sendAnonymousId && !payload.user_id && !payload.email) {
```
Comment (saikumarrs) [inline v0/destinations/intercom/transform.js:147]: We should only be concerned about the user ID.
```suggestion
  if (sendAnonymousId && !payload.user_id) {
```

File: __tests__/data/intercom_output.json [hunk 2]
```diff
@@ -304,8 +307,44 @@
     "files": {},
     "userId": "58b21c2d-f8d5-4410-a2d0-b268a26b7e33"
   },
+  {
+    "statusCode": 400,
+    "error": "Email or userId is mandatory"
+  },
```
Comment (saikumarrs) [inline __tests__/data/intercom_output.json:313]: The test case output data order doesn't align with the inputs.

Comment (shrouti1507) [reply __tests__/data/intercom_output.json:313]: That is another test case, remaining here from previous time. That does not include the new field. I have included a new test case later on. Should we remove this one then?

Comment (saikumarrs) [reply __tests__/data/intercom_output.json:313]: Are all the TCs passing now?

File: v0/destinations/intercom/transform.js (hunk 1 above)
Comment (shrouti1507) [reply v0/destinations/intercom/transform.js:147]: ok

**Review by saikumarrs — APPROVED** [review]

**Pull request closed without merging**